api/src/test/java/org/openmrs/api/OrderServiceTest.java

```diff
@@ -3977,4 +3982,99 @@ public void saveOrder_shouldSaveTheFormNamespaceAndPath() {
 		order = orderService.saveOrder(order, null);
 		assertEquals(NAMESPACE + "^" + FORMFIELD_PATH, order.getFormNamespaceAndPath());
 	}
+
+	@Test
+	public void getAllOrderAttributeTypes_shouldReturnAllOrderAttributeTypes() {
+		assertEquals(4, orderService.getAllOrderAttributeTypes().size());
+	}
+
+	@Test
+	public void getOrderAttributeTypeById_shouldReturnNullIfNoOrderAttributeTypeHasTheProvidedId() {
+		assertNull(orderService.getOrderAttributeTypeById(15));
+	}
+
+	@Test
+	public void getOrderAttributeTypeById_shouldReturnOrderAttributeTypeUsingProvidedId() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeById(2);
+		assertThat(orderAttributeType.getId(), is(2));
+	}
+
+	@Test
+	public void getOrderAttributeTypeByUuid_shouldReturnOrderAttributeTypeUsingProvidedUuid() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeByUuid(
+				"9758d106-79b0-4f45-8d8c-ae8b3f25d72a");
+		assertEquals("Referral", orderAttributeType.getName());
+	}
+
+	@Test
+	public void saveOrderAttributeType_shouldSaveTheProvidedOrderAttributeType()
+			throws ParseException {
+		OrderAttributeType orderAttributeType = new OrderAttributeType();
+		orderAttributeType.setName("Medical Procedures");
+		orderAttributeType.setDatatypeClassname(FreeTextDatatype.class.getName());
+		orderService.saveOrderAttributeType(orderAttributeType);
+		assertNotNull(orderAttributeType.getId());
+	}
+
+	@Test
+	public void saveOrderAttributeType_shouldEditTheExistingOrderAttributeType() {
+		OrderAttributeType orderAttributeType = orderService.getOrderAttributeTypeById(4);
+		assertEquals("Drug", orderAttributeType.getName());
+		orderAttributeType.setName("Drug Dispense");
+		orderService.saveOrderAttributeType(orderAttributeType);
```

Comment:
Would this test fail if you commented out the above line?

Reply:
@dkayiwa please clarify more on "the above line" referenced here. Otherwise, If I am to interpret this correctly, not a good idea to use the scoped variable to test the final outcome...

Reply:
When I comment out the line "orderAttributeType.setName("Drug Dispense");" the OrderAttributeType with Id 4 is not mutated, thus failing the taste. @dkayiwa isn't it the expected behavior? 

 //orderAttributeType.setName("Drug Dispense");

[ERROR] Failures:
[ERROR]   OrderServiceTest.saveOrderAttributeType_shouldEditTheExistingOrderAttributeType:4024 expected: <Drug Dispense> but was: <Drug>

Reply:
By the above line, i mean line 4024

Reply:
When I comment out the line **// orderService.saveOrderAttributeType(orderAttributeType);** and run the test it still passes. 

My suspicion was that since the **orderAttributeType** variable was still in scope, I would be able to get the mutated result after setting it with; **assertEquals("Drug Dispense", orderAttributeType.getName());**

However, the test still passes when I do a new fetch of the **OrderAttributeType** object with; 

assertEquals("Drug Dispense", orderService.getOrderAttributeTypeById(4).getName());

@dkayiwa what is expected here? Since leaving out the line does not really validate the notion of testing that the **saveOrderAttributeType** method actually edits an exiting object without explicitly calling the method on a mutated object.